ceph-backport: rename __atomic_add_unless as atomic_add_unless

This patch was created because of the changes(libceph: a few 
small changes, commit is d3002b974cefbb7c1e325cc296966f768ff76b06 
). The function(__atomic_add_unless) which has be used can't be 
found in the SLE11SP2's kernel source, but we found that 
__atomic_add_unless is just a renamed function of atomic_add_unless 
in the patch(atomic: move atomic_add_unless to generic code, commit 
is f24219b4e90cf70ec4a211b17fbabc725a0ddf3c), so we just need rename 
the function. 

NOTE:
	1) We must review this patch when SLE11SP2's kenrel version is 
	   higher than f24219b4e90cf70ec4a211b17fbabc725a0ddf3c, because
	   the function __atomic_add_unless and atomic_add_unless have 
	   their own meaning after the commit.
	2) All above commit is based on linus's kernel tree.

---

Index: linux-3.0/net/ceph/messenger.c
===================================================================
--- linux-3.0.orig/net/ceph/messenger.c
+++ linux-3.0/net/ceph/messenger.c
@@ -433,7 +433,7 @@ bool ceph_con_opened(struct ceph_connect
  */
 struct ceph_connection *ceph_con_get(struct ceph_connection *con)
 {
-	int nref = __atomic_add_unless(&con->nref, 1, 0);
+	int nref = atomic_add_unless(&con->nref, 1, 0);
 
 	dout("con_get %p nref = %d -> %d\n", con, nref, nref + 1);
 
